### DISCUSSION TITLE
How do you add a favicon to a GitHub page with a theme?

### DISCUSSION THREAD

**Discussion by AlphaZetta** [community/community#23990] [Other Feature Feedback, Questions, & Ideas]
<p>I’m trying to add a favicon to a GitHub page with the ‘slate’ theme but I don’t know where to add it as it doesn’t use traditional html.</p>

**Comment by AlphaZetta** [comment]
<p>It must not require terminal.</p>

**Comment by theNaut1lus** [comment, marked as answer]
<p>Hey <a class="mention" href="/mcloverz">@mcloverz</a>,</p>
<p>Once you have generated the theme, then naviagate to the root directory, you will find an index.html file in your root directory.</p>
<p>after that i believe it should be very straight forward, just add a simple link tag like you would add in your normal static html website.</p>
<p>But just be sure not to put a slash in the beginning when defining the href attribute.</p>
<p>so for example</p>
<pre><code>&lt;link rel="shortcut icon" type="image/x-icon" href="favicon.ico"&gt;
</code></pre>
<p>this assumes you are placing the .ico file in the root directory, if not then,</p>
<pre><code>&lt;link rel="shortcut icon" type="image/x-icon" href="whatever_folder_name/favicon.ico"&gt;
</code></pre>
<p>hope this helps you :slight_smile:</p>
<p>P.S: for assurance purposes i did google what your query was exactly, linking the source from where i got the exact details: <a href="https://medium.com/@LazaroIbanez/how-to-add-a-favicon-to-github-pages-403935604460" rel="nofollow noopener">The Link</a></p>

**Reply by Manw23** [reply]
the author deleted it.

**Comment by AlphaZetta** [comment]
<p>Thanks! I’ll give it a try.</p>

**Comment by AlphaZetta** [comment]
<p>This still doesn’t work as I am using a theme (index.md), not HTML (index.html).</p>

**Comment by AlphaZetta** [comment]
<p>Nevermind, I figured it out! I’ll accept your answer as the solution as it’d probably help most.</p>

**Comment by EthanSK** [comment]
<p>How do you do it</p>

**Comment by steven-tey** [comment]
<p>May I know how did you solve this? I’m asking because I also have an index.md file.</p>

**Comment by olafkotur** [comment]
<p>You can just drag and drop the favicon.ico file into your repository and it’ll work, might need to give it a few minutes to update :slight_smile:</p>

**Comment by annacjacobson** [comment]
<p>I was not able to make this work with the Minimal theme - I tried everything suggested here! I have my favicon saved in my repo as both a .ico and .png file. I also tried adding this code to the top of my index.md file: </p>
<pre><code>&lt;link rel="icon" href="/favicon.ico" type="image/x-icon" /&gt;
</code></pre>
<p>(It works fine when I create an index.html file, but of course that doesn’t work with the theme.)</p>
<p>I also tried adding this to my _config.yml file:</p>
<pre><code>site_favicon: favicon.ico
</code></pre>
<p>Can anyone help me figure out what I’m doing wrong? </p>
<p>EDIT: I realized today that this is a Mac issue (Chrome and Safari). On my work PC, the favicon shows up as it should.</p>

**Comment by zaidi** [comment]
<p>What was the solution? Have been searching for a while, and tried a number of things, but am still unable to figure it out.</p>

**Comment by chengchou** [comment]
<p>I am also using <code>index.md</code> and theme, and I don’t have <code>index.html</code>. I got my <code>favicon</code> work by two steps:</p>
<ol>
<li>Upload the <code>favicon.ico</code> file to my GitHub repository at the root directory.</li>
</ol>
<p>2. Go to <code>_layouts/default.html</code>, and add </p>
<pre><code>&lt;link rel="shortcut icon" type="image/x-icon" href="favicon.ico"&gt;
</code></pre>
<p>into the <code>head</code> (the section surrounded by <code>\&lt;head\&gt;</code> and <code>\&lt;/head\&gt;</code>).</p>
<ol start="3">
<li>Commit the changes.</li>
</ol>

**Comment by Destaq** [comment]
<p>How did you manage to solve it?</p>

**Comment by Kyllingene** [comment]
<p>Following the advice of <a class="mention" href="/chengchou">@chengchou</a> helped me a lot, even though I had no _layouts/ to start with. All I had to do was add a /_layouts/default.html with this code:</p>
<pre><code class="lang-auto">&lt;html&gt;
	&lt;head&gt;
		&lt;link rel="shortcut icon" type="image/x-icon" href="favicon.ico"&gt;
	&lt;/head&gt;
&lt;/html&gt;
</code></pre>
<p>and it displayed my favicon.</p>

**Comment by jroo3121** [comment]
<p>I tried this. Didn’t work.</p>

**Comment by jalloola** [comment]
<p>I tried chengchou’s solution and it didn’t work at first. For some reason it started working when I used a .png file instead of .ico. Worth a try for anyone having troubles! So:</p>

**Comment by JJ1BBY** [comment]
<p><a href="https://github.community/t/favicon-with-github-pages/10242/4" class="inline-onebox">Favicon with github pages - #4 by timrossback</a> tells how to solve this.<br>
default.html will be loaded instead of the index.md.<br>
you can use head.html in the /_layouts.</p>

**Comment by thepwrtank18** [comment]
<p>Does anyone know how to get this to work on the Modernist theme?</p>

**Comment by DulangiK** [comment]
Hi this answer might be bit late, but free feel to try this out, this is the closest can get with shield.io
`[![View  on stackoverflow](https://img.shields.io/stackexchange/stackoverflow/r/12017769)](https://stackoverflow.com/users/12017769/dulangi-kanchana)`


Type this on the index.md file

**Comment by Mr-Saikiran** [comment]
None of this works..

**Comment by ValodokaGames** [comment]
I agree @Mr-Saikiran 
None of this worked for me either. I think your supposed to paste an image code in, but otherwise I don't know if I'm right about that. I tried @chengchou's option, and it didn't work at all. I also tried @theNaut1lus and it didn't work either. Both sadly don't work for me. So what even works?

**Comment by Omegi-05** [comment]
`<head><link rel="icon" type="image/png" href="./omegart.png"></head>`

Hey, try this :
